pkg/ipamd/ipamd.go

```diff
@@ -384,18 +384,40 @@ func (c *IPAMContext) nodeInit() error {
 			time.Sleep(eniAttachTime)
 		}
 	}
-	localPods, err := c.getLocalPodsWithRetry()
+
+	err = c.configureIPRulesForPods(pbVPCcidrs)
 	if err != nil {
-		log.Warnf("During ipamd init, failed to get Pod information from Kubernetes API Server %v", err)
 		ipamdErrInc("nodeInitK8SGetLocalPodIPsFailed")
-		// This can happens when L-IPAMD starts before kubelet.
+		return err
+	}
+	// For a new node, attach IPs
+	increasedPool, err := c.tryAssignIPs()
+	if err == nil && increasedPool {
+		c.updateLastNodeIPPoolAction()
+	} else {
+		return err
+	}
+
+	//Spawning checkAndUpdateRules go-routine
+	go wait.Forever(func() {
+		pbVPCcidrs = c.checkVPCCIDRsAndRules(pbVPCcidrs)
+	}, time.Second)
+
+	return nil
+}
+
+func (c *IPAMContext) configureIPRulesForPods(pbVPCcidrs []string) error {
+	localPods, err := c.getLocalPodsWithRetry()
```

Comment:
This calls the API server. It used to only be done once when a new node starts up, and it's not something we should have to do a lot.

Reply:
I am not sure if I understand this comment. I am relying on this same func to get running Pod IP address for me to re-configure `ip rules` when there is change.